From 9ef185255126b9ed415fa334f585a11c5be4fb1a Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jhrozek@redhat.com>
Date: Mon, 5 Jun 2017 15:19:13 +0200
Subject: [PATCH 86/93] SECRETS: Do not link with c-ares
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Since we started using libcurl for the proxy provider, there is no point
in initializing or linking against c-ares.

If we want to explicitly use a resolver in the future, we should use
libcurl callbacks.

Reviewed-by: Simo Sorce <simo@redhat.com>
Reviewed-by: Fabiano Fidêncio <fidencio@redhat.com>
Reviewed-by: Lukáš Slebodník <lslebodn@redhat.com>
---
 Makefile.am                    | 1 -
 src/responder/secrets/proxy.c  | 2 --
 src/responder/secrets/secsrv.c | 6 ------
 src/responder/secrets/secsrv.h | 3 ---
 4 files changed, 12 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 9fc3dfcd10ab67fa63370fb5b81b93be6bd6400d..273ecc72fba6793b4f46dbb11f6541e2e1bcc930 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1500,7 +1500,6 @@ sssd_secrets_SOURCES = \
     src/util/sss_iobuf.c \
     src/util/tev_curl.c \
     $(SSSD_RESPONDER_OBJ) \
-    $(SSSD_RESOLV_OBJ) \
     $(NULL)
 sssd_secrets_LDADD = \
     $(HTTP_PARSER_LIBS) \
diff --git a/src/responder/secrets/proxy.c b/src/responder/secrets/proxy.c
index a4e97f83ef406e71a1e6509a6b719c47afdfd5b8..a910b38534195e31e3370854b8b9118e8e310d36 100644
--- a/src/responder/secrets/proxy.c
+++ b/src/responder/secrets/proxy.c
@@ -29,7 +29,6 @@
 #define SEC_PROXY_TIMEOUT 5
 
 struct proxy_context {
-    struct resolv_ctx *resctx;
     struct confdb_ctx *cdb;
     struct tcurl_ctx *tcurl;
 };
@@ -585,7 +584,6 @@ int proxy_secrets_provider_handle(struct sec_ctx *sctx,
     pctx = talloc(handle, struct proxy_context);
     if (!pctx) return ENOMEM;
 
-    pctx->resctx = sctx->resctx;
     pctx->cdb = sctx->rctx->cdb;
     pctx->tcurl = tcurl_init(pctx, sctx->rctx->ev);
     if (pctx->tcurl == NULL) {
diff --git a/src/responder/secrets/secsrv.c b/src/responder/secrets/secsrv.c
index b0467e90e65c28929ae0f16b70b58ed9d27560c2..ae2a658ae131e742466796cc47892a234e46f7d3 100644
--- a/src/responder/secrets/secsrv.c
+++ b/src/responder/secrets/secsrv.c
@@ -162,12 +162,6 @@ static int sec_process_init(TALLOC_CTX *mem_ctx,
         goto fail;
     }
 
-    ret = resolv_init(sctx, ev, SEC_NET_TIMEOUT, &sctx->resctx);
-    if (ret != EOK) {
-        /* not fatal for now */
-        DEBUG(SSSDBG_CRIT_FAILURE, "Failed to initialize resolver library\n");
-    }
-
     /* Set up file descriptor limits */
     responder_set_fd_limit(sctx->fd_limit);
 
diff --git a/src/responder/secrets/secsrv.h b/src/responder/secrets/secsrv.h
index 0575cbabab07769d2b17f2c0b815004a11b49f90..1aad272da3ded1a2b3d2d8475ff3f2422c893483 100644
--- a/src/responder/secrets/secsrv.h
+++ b/src/responder/secrets/secsrv.h
@@ -30,10 +30,7 @@
 #include <tevent.h>
 #include <ldb.h>
 
-#define SEC_NET_TIMEOUT 5
-
 struct sec_ctx {
-    struct resolv_ctx *resctx;
     struct resp_ctx *rctx;
     int fd_limit;
     int containers_nest_level;
-- 
2.14.1

